crates/merkle-inclusion/tests/gadget.rs

```diff
@@ -0,0 +1,370 @@
+use bellpepper_core::boolean::{AllocatedBit, Boolean};
+use bellpepper_core::ConstraintSystem;
+
+use bellpepper_core::test_cs::TestConstraintSystem;
+use bellpepper_merkle_inclusion::traits::{GadgetDigest, Keccak, Sha3};
+use bellpepper_merkle_inclusion::{verify_proof, Leaf, Proof};
+use bitvec::order::Lsb0;
+use bitvec::prelude::BitVec;
+use bls12_381::{Bls12, Scalar};
+use digest::{Digest, Output};
+use pairing::Engine;
+
+macro_rules! create_hash_tests {
```

Comment:
Those tests are under-using the trait you're defining. Arguably, once you have the trait defined, macros are overkill for what you want to do with tests here.

You could, for example, do the following:
 ```rust
 fn verify_inclusion_merkle<H: GadgetDigest<Scalar>>() {
     ...
     construct_merkle_tree::<H::OutOfCirCuitHasher>();
     ...
}

#[test]
fn test_verify_inclusion_merkle() {
    verify_inclusion_merkle::<Sha3>();
    verify_inclusion_merkle::<Keccak>();
    ...
}
```

Reply:
Oh yes, completely missed that. Way cleaner!

Reply:
Did so in 6057334.